### PR TITLE
[logs] Add dynamo_timed to get better compilation time breakdown for AOTI 

### PR DESCRIPTION
**Pull request by henrylhtsang**
Adding some dynamo timed for the purpose of better understanding AOTI compilation time.

Probably would require a few more passes. A lot of time is spent in Scheduler.__init__, and not enough annotations are there.

run_command_and_check takes a lot time as well. But there is probably not much we can do. Maybe we can add a config to tune C++ optimization level?

traces:
Screenshot 2024-11-08 at 4 41 10 PM

Differential Revision: D65554141

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang @aakhundov

### CODE REVIEW

File: torch/_inductor/codecache.py [hunk 1]
```diff
@@ -2006,7 +2007,8 @@ def _pad_to_alignment(raw_bytes: bytes) -> bytes:
                     compile_file(input_path, output_o, compile_cmd.split())
                     os.chmod(output_o, 0o644)
                 else:
-                    run_command_and_check(compile_cmd)
+                    with dynamo_timed("run_compile_cmd", log_pt2_compile_event=True):
```
Comment (desertfire) [inline torch/_inductor/codecache.py:2010]: Didn't you already add dynamo_timed to run_command_and_check?

**Comment by desertfire** [comment]
LGTM, but I will let folks who work on compilation time to comment, @masnesral , @jamesjwu

File: torch/_inductor/codecache.py (hunk 1 above)
Comment (henrylhtsang) [reply torch/_inductor/codecache.py:2010]: I think I added them so I can annotate which run_command_and_check it is. But let me think about it more next week. 

Screenshot 2024-11-12 at 2 44 44 PM

**Comment by henrylhtsang** [comment]
@desertfire removed the extra dynamo_tiled outside of run_command_and_check

**Comment by masnesral** [comment]
> LGTM, but I will let folks who work on compilation time to comment, @masnesral , @jamesjwu

Looks fine to me too; the only thing I don't know is if we want all of these in pt2 compile events? @jamesjwu, yes?

**Review by desertfire — APPROVED** [review]

**Pull request closed without merging**